tests/test_pennsieve.py

```diff
@@ -0,0 +1,240 @@
+                "sourceDatasetId": 3,
+                "name": "dataset",
+                "description": "description",
+                "ownerId": 4,
+                "ownerFirstName": "John",
+                "ownerLastName": "Smith",
+                "ownerOrcid": "0000-0000-0000-0000",
+                "organizationName": "organization",
+                "organizationId": 1,
+                "license": "license",
+                "tags": ["tag1", "tag2"],
+                "version": 5,
+                "revision": None,
+                "size": 6,
+                "modelCount": [{"modelName": "model", "count": 0}],
+                "fileCount": 1,
+                "recordCount": 1,
+                "uri": "s3://pennsieve/1/1/",
+                "arn": "arn:aws:s3:::pennsieve/1/1/",
+                "status": "PUBLISH_SUCCEEDED",
+                "doi": "10.00000/abc123",
+                "banner": "https://pennsieve/banner",
+                "readme": "https://pennsieve/dataset-assets/1/1/readme.md",
+                "contributors": [
+                    {
+                        "firstName": "John",
+                        "middleInitial": None,
+                        "lastName": "Smith",
+                        "degree": None,
+                        "orcid": "0000-0000-0000-0000",
+                    }
+                ],
+                "collections": [{"id": 1, "name": "name"}],
+                "externalPublications": [
+                    {"doi": "10.0000/protocols.xx", "relationshipType": "IsSupplementedBy"}
+                ],
+                "sponsorship": {
+                    "title": "title",
+                    "imageUrl": "https://imageurl",
+                    "markup": "markup",
+                },
+                "pennsieveSchemaVersion": "4.0",
+                "createdAt": "2000-12-12",
+                "updatedAt": "2000-12-12",
+                "firstPublishedAt": "2000-12-12",
+                "versionPublishedAt": "2000-12-12",
+                "revisedAt": None,
+                "embargo": False,
+                "embargoReleaseDate": None,
+                "embargoAccess": None,
+            }
+        ],
+    }
+    mocker.patch("pennsieve2.Pennsieve.get", mock_pennsieve.list_datasets)
+    p = PennsieveService(connect=False)
+    actual = p.list_datasets()
+    assert expected == actual
+
+
+def test_list_files(mocker, mock_pennsieve):
+    expected = [
+        {
+            "name": "This is the filename.txt",
+            "datasetId": 1,
+            "datasetVersion": 1,
+            "size": 1,
+            "fileType": "TXT",
+            "packageType": "package",
+            "icon": "Icon",
+            "uri": "s3://pennsieve/a/b/a.txt",
+            "createdAt": None,
+            "sourcePackageId": "N:package:aaaaaa",
+        }
+    ]
+
+    mocker.patch("pennsieve2.Pennsieve.get", mock_pennsieve.list_files)
+    p = PennsieveService(connect=False)
+    print(p.list_files())
```

Comment:
Delete

Reply:
@hsorby I am curious about your opinion on this :-)
I understand that `prints` in code is bad (I would rather use a logger) but in tests I find them very handy since 
- by default they will only show when the test fails
- they are easy to write
- CONS: delays the test

Reply:
* I see them as cluttering any output.
* They are generally important to the author only.
* If they are important then this is a test, the result should be tested not printed.
* It has no long term benefit, no one but the author knows what to expect here.

Reply:
:+1:

Reply:
I actually forgot to remove those prints.  Their main purpose was for debugging output when I didn't get the expected results. I don't think they are needed in the production.